core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -413,6 +415,52 @@ public final void setOverrideConfiguration(ClientOverrideConfiguration overrideC
         overrideConfiguration(overrideConfiguration);
     }
 
+    @Override
+    public final ClientOverrideConfiguration overrideConfiguration() {
+        SdkClientConfiguration configuration = clientConfiguration.build();
+        ClientOverrideConfiguration.Builder builder =  ClientOverrideConfiguration.builder()
+            .executionInterceptors(configuration.option(EXECUTION_INTERCEPTORS))
+            .headers(configuration.option(ADDITIONAL_HTTP_HEADERS))
+            .metricPublishers(configuration.option(METRIC_PUBLISHERS))
+            .executionAttributes(configuration.option(EXECUTION_ATTRIBUTES));
+
+        addOverrideOption(RETRY_POLICY, configuration, builder::retryPolicy);
+        addOverrideAdvancedOption(USER_AGENT_SUFFIX, configuration, builder);
+        addOverrideAdvancedOption(USER_AGENT_PREFIX, configuration, builder);
+        addOverrideOption(API_CALL_TIMEOUT, configuration, builder::apiCallTimeout);
+        addOverrideOption(API_CALL_ATTEMPT_TIMEOUT, configuration, builder::apiCallAttemptTimeout);
+        addOverrideAdvancedOption(DISABLE_HOST_PREFIX_INJECTION, configuration, builder);
+        addOverrideOption(PROFILE_FILE, configuration, builder::defaultProfileFile);
+        addOverrideOption(PROFILE_NAME, configuration, builder::defaultProfileName);
+
+        if (configuration.option(SIGNER_OVERRIDDEN) != null) {
+            builder.putAdvancedOption(SIGNER, configuration.option(SIGNER));
+        }
+
+        if (configuration.option(ENDPOINT_OVERRIDDEN) != null) {
+            builder.putAdvancedOption(ENDPOINT_OVERRIDDEN_OVERRIDE, configuration.option(ENDPOINT_OVERRIDDEN));
+        }
+
+        return builder.build();
+    }
```

Comment:
I'm guessing we're going to forget to update this when we add new settings. Maybe we should change the get/set for override config to just save the object to a field, and then add a mergeExplicitConfig before mergeChildConfig on line 180, and have that do the line 393 to line 411 logic. That way we'll only ever have to update the 'read' logic when we add new fields, and not have to maintain this 'write' logic.